shared/src/question-generators/math/eea.ts

```diff
@@ -0,0 +1,214 @@
+    const { steps } = calculateEEA(a, b)
+    const permalink = serializeGeneratorCall({
+      generator: ExtendedEuclideanAlgorithm,
+      lang,
+      parameters,
+      seed,
+    })
+
+    const table = generateEEATableSteps(a, b, lang, steps)
+
+    const question: MultiFreeTextQuestion = {
+      type: "MultiFreeTextQuestion",
+      path: permalink,
+      name: t(translations, lang, "name"),
+      fillOutAll: true,
+      text: `${t(translations, lang, "eeaQuestion", { a: String(a), b: String(b) })}\n${table}`,
+      feedback: getFeedbackFunction(steps),
+    }
+
+    return { question }
+  },
+}
+
+function getFeedbackFunction(
+  steps: Array<{
+    dividend: number
+    divisor: number | null
+    quotient: number | null
+    remainder: number
+    s: number
+    t: number
+  }>,
+): MultiFreeTextFeedbackFunction {
+  return ({ text }) => {
+    let allStepsCorrect = true
+
+    const feedbackDetails = steps
+      .filter((step) => step.quotient !== null) // skip null line
+      .map((step, index) => {
+        const stepIndex = index + 1
+
+        // parse and compare user inputs for each field
+        const userQuotient = parseInt(text[`quotient${stepIndex}`], 10)
+        const userRemainder = parseInt(text[`remainder${stepIndex}`], 10)
+        const quotientCorrect = userQuotient === step.quotient
+        const remainderCorrect = userRemainder === step.remainder
+
+        // exclude checks for filled in fields for the first row
+        let dividendCorrect = true
+        let divisorCorrect = true
+        //  include for subsequent rows
+        if (index > 0) {
+          const userDividend = parseInt(text[`dividend${stepIndex}`], 10)
+          const userDivisor = parseInt(text[`divisor${stepIndex}`], 10)
+          dividendCorrect = userDividend === step.dividend
+          divisorCorrect = userDivisor === step.divisor
+        }
+
+        // correctness for the steps
+        if (!dividendCorrect || !divisorCorrect || !quotientCorrect || !remainderCorrect) {
+          allStepsCorrect = false
+        }
+
+        return `${stepIndex}: $ ${step.dividend} = ${step.quotient} \\cdot  ${step.divisor} + ${step.remainder}$\\n `
+      })
+      .join("; ")
+
+    // check final coefficients for linear combination
+    const initialStep = steps[0]
+    const a = initialStep.dividend
+    const b = initialStep.divisor
+    const finalStep = steps[steps.length - 1]
+    const gcd = finalStep.remainder
+    const userCoefA = parseInt(text["coefA"], 10)
+    const userCoefB = parseInt(text["coefB"], 10)
+    const coefficientsCorrect = userCoefA === finalStep.s && userCoefB === finalStep.t
+
+    // correctness for steps and coefficients
+    const isCorrect = allStepsCorrect && coefficientsCorrect
+
+    return {
+      correct: isCorrect,
+      correctAnswer: isCorrect
+        ? ""
+        : `${feedbackDetails}\\n$ \\text{gcd}(${a}, ${b}) = ${gcd} = ${finalStep.s} \\cdot ${a} + ${finalStep.t} \\cdot ${b} $`,
```

Comment:
Can be simplified to `correctAnswer: `${feedbackDet...`. If the user answer is correct, we don't display a correct answer.